core/src/components/action-sheet/action-sheet-interface.ts

```diff
@@ -8,7 +8,15 @@ export interface ActionSheetOptions {
   backdropDismiss?: boolean;
   translucent?: boolean;
   animated?: boolean;
+  /**
+   * @deprecated To change the default appearance of the popover, use the `theme` option.
```

Comment:
Talked with Liam, we will pull this deprecation back for now. I'll need to add additional runtime validations for now to make sure developers are not using an invalid `mode` and `theme` configuration. 

This applies to all controller overlays.

Reply:
I don't believe configuring `mode` makes sense from the overlay controllers. That is what the "for now" refers to. We will do further exploration if developers really only need to configure the visual appearance and would use `theme` in the future to do that. Changing the platform behavior should be something developers rarely or never need to do. 

Overlay controllers have always had more structured and limited APIs. The advantage of _not_ exposing `mode` is that we don't need additional JS overhead of validating configurations.

I'm still working on the runtime validations locally. Once that is ready I will push that work up, along with the updated overlay interfaces and deprecation removals.

Reply:
Doesn't this mean that developers can no longer change the `mode` on controller overlays, but they still can on inline ones? Wouldn't we want the two APIs to be as in sync as possible?

Reply:
Nevermind, just noticed you put the mode prop back through `OverlayOptions` 👍